### PR TITLE
Code quality push (for platinum upstream)

### PR DESCRIPTION
**Pull request by CoMPaTech**
Including #129 as requested by @bouwew 

# Changes

- Availability (as per #129) handling
- Introduce `ATTR_ENABLED_DEFAULT` and included on the Entity classes
- Rewrite gateway sensoric maps to dictionaries as with stick - superceeding introduced `CUSTOM_ICONS`
- Handle icon only if no DEVICE_CLASS is set (as per Core decision)
- Set `PARALLEL_UPDATES` on `sensor` and `binary_sensor` as first included platforms
- Import more ATTR consts from core

# Checklist:

**Silver**

- [x] Connection/configuration is handled via a component.
- Yes, using the 'hub' approach and DataUpdateCoordinator to talk to our module
- [x] Set an appropriate SCAN_INTERVAL (if a polling integration)
- Yes, dependent on type of hub (e.g. climate devices every minute, DSMR every 10 seconds)
- [x] Raise PlatformNotReady if unable to connect during platform setup (if appropriate)
- Yes, handled appropriately
- [ ] Handles expiration of auth credentials. Refresh if possible or print correct error and fail setup. If based on a config entry, should trigger a new config entry flow to re-authorize.
- n/a there is a static password printed on the back of the devices
- [ ] Handles internet unavailable. Log a warning once when unavailable, log once when reconnected.
- n/a local devices
- [x] Handles device/service unavailable. Log a warning once when unavailable, log once when reconnected.
- Per 'hub' using DataUpdateCoordinator
- [x] Set available property to False if appropriate
- Per 'hub' using DataUpdateCoordinator with UpdateFailed
- [x] Entities have unique ID (if available)
- IDs are based on the unique ID of the 'hub' (i.e. Smile) and the unique id's in the API data, where appropriate a descriptive string has been added.

**Gold**

- [x] Configurable via config entries.
- Yes, implemented accordingly
- [x] Don't allow configuring already configured device/service (example: no 2 entries for same hub)
- Checking the unique_ids
- [x] Tests for the config flow
- Covered fully as per requirement for PRs nowadays
- [x] Discoverable (if available)
- Yes, implemented accordingly
- [x] Set unique ID in config flow (if available)
- Yes, implemented accordingly
- [x] Entities have device info (if available)
- Yes, implemented accordingly
- [x] Tests for fetching data from the integration and controlling it
- Yes, using (selected) fixtures generated from actual data also used for testing our module
- [x] Has a code owner
- @bouwew and @CoMPaTech (and soon including @brefra)
- [x] Entities only subscribe to updates inside async_added_to_hass and unsubscribe inside async_will_remove_from_hass
- Yes, implemented accordingly
- [x] Entities have correct device classes where appropriate
- Yes, where available
- [x] Supports entities being disabled and leverages Entity.entity_registry_enabled_default to disable less popular entities
- We already limit the amount of entities being created using the dicts in the consts. We have implemented the `enabled_default` structure into the dicts on const to allow for this!
- Plugwise actually has lots more available not deemed neccesary by our testing-community nor recent HA-core issue requests.
- [ ] If the device/service API can remove entities, the integration should make sure to clean up the entity and device registry.
- Our module does not support removal of entities therefor no service has been implemented in the integration yet

**Platinum**

- [x] Set appropriate PARALLEL_UPDATES constant
- Included in `binary_sensor` and `sensor` as these are the first platforms loaded through our platforms_list
- [x] Support config entry unloading (called when config entry is removed)
- Yes, implemented accordingly
- [x] Integration + dependency are async
- Yes, implemented accordingly
- [x] Uses aiohttp and allows passing in websession (if making HTTP requests)
- Yes, implemented accordingly

### CODE REVIEW

**Comment by CoMPaTech** [comment]
⚠️ TODO: Let's not forget: `illuminance` in `THERMOSTAT_SENSORS` should be returned to `True`ly being enabled ... disabled that for testing purposes with Anna :)

File: custom_components/plugwise/const.py [hunk 1]
```diff
@@ -82,12 +80,11 @@
 NO_NOTIFICATION_ICON = "mdi:mailbox-outline"
 SWITCH_ICON = "mdi:electric-switch"
 
-# __init__ consts:
+# gatewau consts:
```
Comment (bouwew) [inline custom_components/plugwise/const.py:83]: Typo: gateway

File: custom_components/plugwise/const.py [hunk 2]
```diff
@@ -108,138 +105,107 @@
...
-ATTR_BATTERY_LEVEL = ["Charge", PERCENTAGE, DEVICE_CLASS_BATTERY]
-ATTR_ILLUMINANCE = ["Illuminance", UNIT_LUMEN, DEVICE_CLASS_ILLUMINANCE]
-ATTR_PRESSURE = ["Pressure", PRESSURE_BAR, DEVICE_CLASS_PRESSURE]
+def temperature_sensor(name_default = "Temperature", enabled_default = True): 
+    """Return standardized dict with temperature description."""
+    return {
+        ATTR_DEVICE_CLASS: DEVICE_CLASS_TEMPERATURE,
+        ATTR_ENABLED_DEFAULT: enabled_default,
+        ATTR_ICON: None,
+        "name": name_default,
```
Comment (bouwew) [inline custom_components/plugwise/const.py:114]: Import ATTR_NAME

File: custom_components/plugwise/const.py [hunk 3]
```diff
@@ -261,123 +227,123 @@
...
 SENSORS = {
     AVAILABLE_SENSOR_ID: {
-        "class": None,
-        "enabled_default": False,
-        "icon": "mdi:signal-off",
+        ATTR_DEVICE_CLASS: None,
+        ATTR_ENABLED_DEFAULT: False,
+        ATTR_ICON: "mdi:signal-off",
         "name": "Available",
         "state": "get_available",
```
Comment (bouwew) [inline custom_components/plugwise/const.py:234]: Import ATTR_STATE

File: custom_components/plugwise/sensor.py [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 import logging
 
-from homeassistant.const import ENERGY_KILO_WATT_HOUR, PERCENTAGE
+from homeassistant.const import ENERGY_KILO_WATT_HOUR, PERCENTAGE, ATTR_UNIT_OF_MEASUREMENT, ATTR_DEVICE_CLASS, ATTR_ICON
```
Comment (bouwew) [inline custom_components/plugwise/sensor.py:5]: Black formatting

File: custom_components/plugwise/climate.py [hunk 5]
```diff
@@ -77,6 +77,8 @@ def __init__(
         self, api, coordinator, name, dev_id, loc_id, model, min_temp, max_temp
     ):
         """Set up the Plugwise API."""
+        self._enabled_default = True
+
```
Comment (bouwew) [inline custom_components/plugwise/climate.py:81]: Where is the rest of the enabled_default code?

File: custom_components/plugwise/const.py (hunk 1 above)
Comment (CoMPaTech) [reply custom_components/plugwise/const.py]: Oops

File: custom_components/plugwise/const.py (hunk 2 above)
Comment (CoMPaTech) [reply custom_components/plugwise/const.py]: Done

File: custom_components/plugwise/const.py (hunk 3 above)
Comment (CoMPaTech) [reply custom_components/plugwise/const.py]: Done

File: custom_components/plugwise/sensor.py (hunk 4 above)
Comment (CoMPaTech) [reply custom_components/plugwise/sensor.py]: Blacked everything in latest commit (tnx!)

File: custom_components/plugwise/climate.py (hunk 5 above)
Comment (CoMPaTech) [reply custom_components/plugwise/climate.py]: I removed this line now, lets keep the conversation open - if I added that here (for some reason) the climate entity wasn't here at all, but I don't recall if that was due to enabled in the SmileGateway entity or it being here (maybe we should just add it back in again).

**Review by bouwew — APPROVED** [review]
LGTM!

File: custom_components/plugwise/climate.py (hunk 5 above)
Comment (bouwew) [reply custom_components/plugwise/climate.py]: When you give climate the same treatment as the other platforms, they become disable-able?
And when you leave out all enabled_default-code then the climate-entities are always there?

I'd say, no need for disabling climate-entities :)

Comment (CoMPaTech) [reply custom_components/plugwise/climate.py]: I'll give it a whirl tonight, for now your last line was sort-of the quickfix :P

**Comment by CoMPaTech** [comment]
> ⚠️ TODO: Let's not forget: `illuminance` in `THERMOSTAT_SENSORS` should be returned to `True`ly being enabled ... disabled that for testing purposes with Anna :)

note2self

File: custom_components/plugwise/climate.py (hunk 5 above)
Comment (CoMPaTech) [reply custom_components/plugwise/climate.py]: Lets stick to 'not disabling the climate entity' itself, it makes sense

File: custom_components/plugwise/binary_sensor.py [hunk 6]
```diff
@@ -174,9 +178,15 @@ async def async_setup_entry_gateway(hass, config_entry, async_add_entities):
...
+    def __init__(
+        self, api, coordinator, name, dev_id, enabled_default, binary_sensor, model
+    ):
         """Set up the Plugwise API."""
-        super().__init__(api, coordinator, name, dev_id, binary_sensor)
+        self._enabled_default = enabled_default, True
+
+        super().__init__(
+            api, coordinator, name, dev_id, self._enabled_default, binary_sensor
+        )
```
Comment (brefra) [inline custom_components/plugwise/binary_sensor.py:189]: `self._enabled_default` will be a `Tuple` and is passed into `super()`. To me that does not seems to be right.

File: custom_components/plugwise/sensor.py [hunk 7]
```diff
@@ -207,12 +221,17 @@ class GwThermostatSensor(SmileSensor, Entity):
 
     def __init__(self, api, coordinator, name, dev_id, sensor, sensor_type):
         """Set up the Plugwise API."""
-        super().__init__(api, coordinator, name, dev_id, sensor)
+        _LOGGER.error("HOI sensor %s", sensor)
+        _LOGGER.error("HOI type   %s", sensor_type)
+        self._enabled_default = sensor_type["enabled_default"]
```
Comment (brefra) [inline custom_components/plugwise/sensor.py:226]:
```suggestion
        self._enabled_default = sensor_type[ATTR_ENABLED_DEFAULT]
```

File: custom_components/plugwise/sensor.py [hunk 8]
```diff
@@ -272,15 +292,19 @@ class GwPowerSensor(SmileSensor, Entity):
...
         if model is None:
-            self._model = sensor_type[SENSOR_MAP_MODEL]
+            self._model = sensor
 
-        self._unit_of_measurement = sensor_type[SENSOR_MAP_UOM]
-        self._dev_class = sensor_type[SENSOR_MAP_DEVICE_CLASS]
+        self._unit_of_measurement = sensor_type[ATTR_UNIT_OF_MEASUREMENT]
+        self._dev_class = sensor_type[ATTR_DEVICE_CLASS]
+        if not self._dev_class:
+            self._icon = sensor__type[ATTR_ICON]
```
Comment (brefra) [inline custom_components/plugwise/sensor.py:307]:
```suggestion
            self._icon = sensor_type[ATTR_ICON]
```

File: custom_components/plugwise/sensor.py [hunk 9]
```diff
@@ -207,12 +221,17 @@ class GwThermostatSensor(SmileSensor, Entity):
...
-        self._model = sensor_type[SENSOR_MAP_MODEL]
-        self._unit_of_measurement = sensor_type[SENSOR_MAP_UOM]
-        self._dev_class = sensor_type[SENSOR_MAP_DEVICE_CLASS]
+        super().__init__(api, coordinator, name, dev_id, self._enabled_default, sensor)
+
+        self._model = sensor
+        self._unit_of_measurement = sensor_type[ATTR_UNIT_OF_MEASUREMENT]
+        self._dev_class = sensor_type[ATTR_DEVICE_CLASS]
+        if not self._dev_class:
+            self._icon = sensor__type[ATTR_ICON]
```
Comment (brefra) [inline custom_components/plugwise/sensor.py:234]:
```suggestion
            self._icon = sensor_type[ATTR_ICON]
```

File: custom_components/plugwise/binary_sensor.py (hunk 6 above)
Comment (CoMPaTech) [reply custom_components/plugwise/binary_sensor.py:189]: True, wrong copy/paste remnant (well spotted)

**Pull request merged**